todos los cambios estan en el archivo actualizacion 0.1.1.txt

zwisky opened the pull request:
<3

zwisky committed:
todos los cambios estan en el archivo actualizacion 0.1.1.txt

MattRomero commented on actualizacion 0.1.1.txt:
```diff
@@ -0,0 +1,8 @@
+actualizacion npm update.
+actualizacion Node.js v10.9.2 Actualiza el proyecto para que sea compatible con Node.js v23.6.0.
+"rank.js" se modifico los nombres de los chars, respuesta a errores en la consola y la llamada a la API; loadChars, getChar, rankChar.
+"index.js" se actualizo import react-dom
+"package.json" se actualizo react, react-dom y react-script
+actualizacion plugin babel/plugin-proposal-private-property-in-object"
+cambios a tema oscuro en "styles.css"
+modificacion del icon en index.html
```
Esto va en la descripción del PR, no es necesario crear un archivo aparte

MattRomero commented on package.json:
```diff
@@ -41,6 +41,7 @@
     ]
   },
   "devDependencies": {
+    "@babel/plugin-proposal-private-property-in-object": "^7.21.11",
```
Esta dependencia se instaló con las actualizaciones?

zwisky replied:
esta dependencia es necesaria para resolver un conflicto en el terminal que arroja problemas críticos y moderados que podrían afectar la seguridad o funcionalidad de tu proyecto. El terminal sugiere usar npm audit fix --force, pero esto puede introducir cambios importantes (breaking changes).

las recomendación es instalar bootstrap@5.3.3
pero al instalar esta dependencia ya no arroja el error

MattRomero commented on src/components/rank.js:
```diff
@@ -9,21 +9,21 @@ export class Rank extends React.Component {
-        "Dimitrij+del+Toro",
-        "King+Nakio",
-        "Loko+Zwisky",
+        "Dimitrij del Toro", // Cambia + por espacios normales
```
El comentario de "Cambio + por espacios normales" es innecesario, simplemente mencionalo en la descripción del PR, y lo cambias acá

MattRomero commented on src/components/rank.js:
```diff
@@ -38,37 +38,66 @@ export class Rank extends React.Component {
+      return [];
+    }
+  
+    loadedCharList.sort((a, b) => b.level - a.level); // Ordena por nivel
```
Spanglish ajajsfd dejalo en inglés mejor o lo borras

MattRomero commented on src/components/rank.js:
```diff
@@ -38,37 +38,66 @@ export class Rank extends React.Component {
+      }
+      let data = await response.json();
+  
+      if (data && data.character && data.character.character) {
```
Esto es lo mismo que

```suggestion
      if (data?.character?.character) {
```

Si es que alguno de los que tienen el ? no existe, devuelve undefined y no tira error, no se cumple la condición

MattRomero commented on src/components/rank.js:
```diff
@@ -38,37 +38,66 @@ export class Rank extends React.Component {
 
   rankChars = (charList) => {
-    let previousCharLevel = charList[0]['level'];
+    if (!charList || charList.length === 0) {
```
```suggestion
    if (charList?.length === 0) {

```

MattRomero commented on src/index.js:
```diff
@@ -17,7 +17,7 @@ function App() {
       </Row>
       <Row className="align-items-center">
-       <span className="dot"></span>Sharing partners
+        <span className="dot"></span>Sharing partners
```
Esto no cambió nada jaja, por qué sale como cambio?

zwisky replied:
hay un espacio xd mala mia

MattRomero commented on src/index.js:
```diff
@@ -27,4 +27,5 @@ function App() {
 }
 
 const rootElement = document.getElementById("root");
+// Actualizacion de Version React 16.12.0 a React 16.14.0
```
Comentario innecesario

zwisky committed:
actualizacion npm update.
actualizacion Node.js v10.9.2 Actualiza el proyecto para que sea compatible con Node.js v23.6.0.
"rank.js" se modifico los nombres de los chars, respuesta a errores en la consola y la llamada a la API; loadChars, getChar, rankChar.
"index.js" se actualizo import react-dom
"package.json" se actualizo react, react-dom y react-script
actualizacion plugin babel/plugin-proposal-private-property-in-object" (solución de error en el terminal, por version del bootstrap)
cambios a tema oscuro en "styles.css"
modificacion del icon en index.html

actualizacion a version 0.1.1
borre todos los comentarios innecesarios
modifique el codigo if (data && data.character && data.character.character) por if (data?.character?.character)
